packages/firestore/src/util/input_validation.ts

```diff
@@ -292,6 +292,31 @@ export function validateNamedOptionalPropertyEquals<T>(
   }
 }
 
+/**
+ * Validates that the provided argument is part of the
+ *
+ * @param functionName Function making the validation call.
+ * @param enums Array containing all possible values for the enum.
+ * @param enumName Custom type name.
+ * @param position Position of the argument in `functionName`.
+ * @param argument Arugment to validate.
+ */
+export function validateArgEnum<T>(
```

Comment:
I'd actually drop "Arg" (most of our validation functions don't have it).  This also isn't technically an enum (it's a string union or whatever), though I'm not wild about validateStringUnion().  Maybe `validateStringEnum()`?